man/special-symbols.Rd

```diff
@@ -66,5 +68,25 @@ DT[{cat(sprintf('in i, .N is \%d\n', .N)); a < .N/2},
 # .I can be different in j and by, enabling rowwise operations in by
 DT[, .(.I, min(.SD[,-1]))]
 DT[, .(min(.SD[,-1])), by=.I]
+
+# These examples illustrate the importance of using copy(.BY) when capturing .BY values in a list outside the data.table context to avoid unexpected behavior.
+dt <- data.table(l = letters[1:5], b = rnorm(5))
+setkey(dt, l)
+
+## Incorrect usage
+byg <- list()
+dt[, m := {
+  byg <<- append(byg, .BY)  # Incorrect: Missing copy(.BY)
+  mean(b)
+}, by = l]
+str(byg)  # All elements of 'byg' will contain the .BY value of the last group
+
+## Correct usage
+byg <- list()
+dt[, m := {
```

Comment:
these examples are perhaps too long.
If I understand correctly, the important different is .BY versus copy(.BY) right?
So can you please delete as much other code (:= append, mean)  as possible to make the example simpler? ideally one line instead of 6 for both incorrect/correct usage